Fix/gosec linters

cwarnerdev opened the pull request.

kape1395 commented on packages/vm/core/governance/governanceimpl/impl.go:
```diff
@@ -55,5 +64,9 @@ func initialize(ctx iscp.Sandbox) (dict.Dict, error) {
 		state.Set(governance.VarFeeColor, codec.EncodeColor(feeColor))
 	}
+
+	collections.NewMap(state, governance.VarAccessNodeCandidates).Erase()
+	collections.NewMap(state, governance.VarAccessNodes).Erase()
+	collections.NewArray16(state, governance.VarValidatorNodes).MustErase()
```
I wanted to have the state variables and their types mentioned here explicitly.
Such call will actually do nothing. Maybe there is a better way to declare variables and their intended data types?

dessaya replied:
We don't have a standard way of doing that. What I like to do is having functions like:

```
func getAccessNodeCandidatesMap(state kv.KVStore) *collections.Map {
  return collections.NewMap(state(), governance.VarAccessNodeCandidates)
}

func getAccessNodeCandidatesMapR(state kv.KVStoreReader) *collections.ImmutableMap {
  return collections.NewMapReadOnly(state(), governance.VarAccessNodeCandidates)
}

func getAccessNodesMap(state kv.KVStore) *collections.Map {
  return collections.NewMap(state(), governance.VarAccessNodes)
}

func getAccessNodesMapR(state kv.KVStoreReader) *collections.ImmutableMap {
  return collections.NewMapReadOnly(state(), governance.VarAccessNodes)
}

func getValidatorNodesArray(state kv.KVStore) *collections.Array16 {
  return collections.NewArray16(state(), governance.VarValidatorNodes)
}

func getValidatorNodesArrayR(state kv.KVStoreReader) *collections.ImmutableArray16 {
  return collections.NewArray16ReadOnly(state(), governance.VarValidatorNodes)
}
```

kape1395 replied:
ok, removed those `Erase` here.

kape1395 commented on packages/vm/core/governance/governanceimpl/accessnodesImpl.go:
```diff
@@ -0,0 +1,119 @@
+// Can only be invoked by the chain owner.
+//
+//  changeAccessNodes(
+//    actions: map(pubKey => {0:remove, 1:accept, 2:drop})
```
Good point. Will change.

jorgemmsilva commented on packages/vm/core/governance/governanceimpl/accessnodesImpl.go:
```diff
@@ -0,0 +1,117 @@
+	params := kvdecoder.New(ctx.Params(), ctx.Log())
+	paramCandidate := params.MustGetBool(governance.ParamCandidateNodeCandidate, true)
+
+	ani := governance.AccessNodeInfo{
+		NodePubKey:    params.MustGetBytes(governance.ParamCandidateNodePubKey),
+		ValidatorAddr: ctx.Request().SenderAddress().Bytes(), // Not from params, to have it validated.
+		Certificate:   params.MustGetBytes(governance.ParamCandidateNodeCertificate),
+		ForCommittee:  params.MustGetBool(governance.ParamCandidateNodeForCommittee, false),
+		AccessAPI:     params.MustGetString(governance.ParamCandidateNodeAccessAPI, ""),
+	}
```
suggestion: we could receive the entire `AccessNodeInfo` as a parameter, right? no need to split each property into a separate param.

kape1395 replied:
Yes, I was thinking about that option. The structure is not exactly the same. The parameters don't include the ValidatorAddr field.

Another possible problem is that the parameter names are then hidden from other parameters, thus making it easier to make them coinciding. E.g. `ParamCandidateNodeCandidate` would be defined in one place (function params), and other parameters in the AccessNodeInfo structure. In the dictionary they would be on the same level. Not sure, which way is better.

kape1395 replied:
Updated to use the AccessNodeInfo struct for params, revoke function extracted to a separate one.

jorgemmsilva commented on packages/vm/core/governance/governanceimpl/impl.go:
```diff
@@ -29,6 +32,11 @@ var Processor = governance.Contract.Processor(initialize,
+
+	// access nodes.
+	governance.FuncGetChainNodes.WithHandler(getChainNodesFuncHandler),
+	governance.FuncCandidateNode.WithHandler(candidateNodeFuncHandler),
```
its kinda hard to tell what this function does by name: `candidateNodeFuncHandler`
I propose changing the name to something more descriptive like: `addCandidateNode`

kape1395 replied:
Renamed.

cwarnerdev committed:
Use safecast for safe uint32 conversions.

Replaced direct uint32 conversions with safecast.MustConvert to ensure safe and reliable type casting. Updated both map.go and array.go to utilize this approach consistently.

cwarnerdev committed:
Refactor NodeID logic to use safe type conversion

Replaced direct type conversion with `safecast.MustConvert` for better safety and reliability. This ensures type conversions are always validated, reducing potential runtime errors.

cwarnerdev committed:
Add safe type conversions using safecast library

Replaced direct type conversions with safecast to prevent potential overflows and ensure safe handling of variables across the codebase. Error handling has been added for cases where conversions may fail, improving system robustness.
